lib/wine_o_inventory_api_web/plugs/check_json_spec.ex

```diff
@@ -0,0 +1,26 @@
+defmodule WineOInventoryApiWeb.Plugs.CheckJsonSpec do
+    use WineOInventoryApiWeb, :controller
+    import Plug.Conn
+
+    def init(_params) do
+    end
+
+    def call(conn, _params) do
+        inspect_json(conn.method, conn)
+    end
+
+    def inspect_json("POST", conn) do
+        request_body = conn.body_params
+            if (request_body["data"] == nil || request_body["data"]["attributes"] == nil) do
+                conn
+                |> put_status(:bad_request)
+                |> json(%{errors: %{detail: "Bad Request"}})
+                |> halt()
+            end
+        conn
+    end
+
+    def inspect_json(_method, conn) do
+        conn
+    end
+end
```

Comment:
This file also needs a test.

Reply:
I added a test to check if an error of 400 is sent when the request POST body doesn't match the JSON API spec, well at least the portion that I am enforcing. I haven't been able to figure out how to test the case when everything checks out correctly, such that the request POST body matches the JSON API spec.